packages/functional-tests/pages/baseTokenCode.ts

```diff
@@ -0,0 +1,38 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { BaseLayout } from './layout';
+import { expect } from '../lib/fixtures/standard';
+
+export class BaseTokenCodePage extends BaseLayout {
+  // This convention seems pretty useful. Maybe we should put it on BaseLayout.
+  checkForPath(path?: string) {
+    if (path) {
+      expect(this.page.url()).toContain(path);
+    }
+  }
+
+  get tooltip() {
+    this.checkForPath(this.path);
+    return this.page.locator('.tooltip');
+  }
+
+  get successMessage() {
+    return this.page.locator('.success');
```

Comment:
In my opinion just having the locator inline is more clear. I don't see the benefit of having them in a map somewhere. They really shouldn't be referenced over and over again anyways.